CircuitPython_Slider/code.py.py

```diff
@@ -0,0 +1,347 @@
+            #1 means that the camera is next to the motor
+            if z == 1:
+                b_state = None
+                time.sleep(2)
+                #changes states
+                pause = 0
+                stop = 0
+                z = 0
+        #a is defined to slide in reverse of the prev. slide
+        #only active if the slider is in the 'stopped' state
+        if buttons.a and a_state == "pressed" and stop == 1:
+            #z defines location of the camera on the slider
+            #1 means that the camera is next to the motor
+            if z == 1:
+                a_state = None
+                time.sleep(2)
+                stop = 0
+                pause = 0
+                #2 allows the 'regular' slide loop to run
+                #as if the 'select' button has been pressed
+                z = 2
+            #0 means that the camera is opposite the motor
+            if z == 0:
+                a_state = None
+                #same script as the 'regular' slide loop
+                time.sleep(2)
+                display.show(progBarGroup)
+                press = time.monotonic()
+                print(press)
+                text_area.text = slide_begin[mode]
+                onOff += 1
+                pause = 0
+                stop = 0
+                if onOff > 1:
+                    onOff = 0
+                for i in range(17200):
+                    start = time.monotonic()
+                    real_time = start - press
+                    end = slide_duration[mode] - real_time
+                    mins_remaining = end / 60
+                    if mins_remaining < 0:
+                        mins_remaining += 60
+                    total_sec_remaining = mins_remaining * 60
+                    mins_remaining, total_sec_remaining = divmod(end, 60)
+                    #only difference is that the motor is stepping backwards
+                    kit.stepper1.onestep(direction=stepper.BACKWARD, style=stepper.MICROSTEP)
+                    time.sleep(speed[mode])
+                    if i == slide_checkin[check]:
+                        print("0%d:%d" %
+                              (mins_remaining, total_sec_remaining))
+                        if total_sec_remaining < 10:
+                            text_area.text = "%d:0%d" % (mins_remaining, total_sec_remaining)
+                        else:
+                            text_area.text = "%d:%d" % (mins_remaining, total_sec_remaining)
+                        check = check + 1
+                    if check > 19:
+                        check = 0
+                    if end < 10:
+                        display.show(stopGroup)
+                #state changes
+                kit.stepper1.release()
+                pause = 1
+                onOff = 0
+                stop = 1
+                z = 1
+                check = 0
+                time.sleep(2)
+                display.show(reverseqGroup)
+    except MemoryError:
```

Comment:
??? this is kinda weird. are you getting memory errors??

Reply:
Agreed that this is a bit weird. If you're experiencing memory issues, you may want to consider using OnDiskBitmap instead of the Image Load library. See here: https://learn.adafruit.com/pyportal-calculator-using-the-displayio-ui-elements/ui-quickstart

Reply:
Apologies, that was from when I first started working on the code. I haven't had any memory errors, will take it out.